### PR TITLE
feat(zfspv): remove finalizer that is owned by ZFS-LocalPV

### PR DESCRIPTION
**Pull request by pawanpraka1**
fixes: https://github.com/openebs/zfs-localpv/issues/302

Signed-off-by: Pawan <pawan@mayadata.io>

**Why is this PR required? What issue does it fix?**:

User might want to add extra protection on the ZFSVolume CR and wants to add finalizers on it. The current code does not allow the user to setup the user's finalizers on the ZFSVolume CR.
 
**What this PR does?**:

We set the Finalizer to nil while handling the delete event, instead,
we should try to destroy the volume when there are no user finalizers
set. User might have added his own finalizers and we should not try to destroy
the volumes util those user finalizers are removed.

**Does this PR require any upgrade changes?**:

PR handles upgrade scenario.

### CODE REVIEW

**Review by akhilerm — COMMENTED** [review]
@pawanpraka1 We need to make this same change in LVM localpv right?

File: pkg/zfs/volume.go [hunk 1]
```diff
@@ -358,3 +350,32 @@ func UpdateRestoreInfo(rstr *apis.ZFSRestore, status apis.ZFSRestoreStatus) erro
...
+	for _, fin := range finalizers {
+		if fin != ZFSFinalizer {
+			userFin = append(userFin, fin)
+		}
+	}
+	return userFin
+}
+
+// IsVolumeReady returns true if volume is Ready
+func IsVolumeReady(vol *apis.ZFSVolume) bool {
```
Comment (akhilerm) [inline pkg/zfs/volume.go:367]: :+1:

**Comment by pawanpraka1** [comment]
> We need to make this same change in LVM localpv right?

@akhilerm yes, it is good to have this change in LVM LocalPV repo also.

File: pkg/zfs/volume.go [hunk 2]
```diff
@@ -290,10 +286,6 @@ func UpdateSnapInfo(snap *apis.ZFSSnapshot) error {
 	finalizers := []string{ZFSFinalizer}
 	labels := map[string]string{ZFSNodeKey: NodeID}
 
-	if snap.Finalizers != nil {
```
Comment (akhilerm) [inline pkg/zfs/volume.go:293]: Why was this removed? If user has added a custom finalizer and snapshot is not ready, will this mark the snapshot CR as ready?

Comment (pawanpraka1) [reply pkg/zfs/volume.go:293]: if this check is there, snapshot will never be ready, so we have to remove as it was assumed that no one will create the resources manually.

**Review by akhilerm — APPROVED** [review]
LGTM
